pkg/templates/git.go

```diff
@@ -456,3 +461,46 @@ func getValidVersions(
 
 	return validVersions, versionSchema, nil
 }
+
+// SyncGitTrustedCA syncs git trusted CA from settings,
+// and sets environment variable GIT_SSL_CAINFO.
+func SyncGitTrustedCA(ctx context.Context, m settingbus.BusMessage) error {
+	for i := 0; i < len(m.Refers); i++ {
+		if m.Refers[i] == nil || m.Refers[i].Name == "" {
+			continue
+		}
+
+		if m.Refers[i].Name != settings.SSLTrustedCAFile.Name() {
+			continue
+		}
+
+		err := SetGitCAEnvVar(ctx, m.TransactionalModelClient)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// SetGitCAEnvVar sets environment variable GIT_SSL_CAINFO from settings.
```

Comment:
Does it work cloning from github.com when GIT_SSL_CAINFO is set?

Reply:
Yes, setting `GIT_SSL_CAINFO` extends the CA certificate trust chain used by Git, but it doesn't override the system's default certificates. GitHub's SSL certificates are signed by a public CA and are generally trusted by default.